src/Form/CloudFlareAdminSettingsForm.php

```diff
@@ -2,20 +2,100 @@
 
 /**
  * @file
- * Contains Drupal\cloudflare\Form\DefaultForm.
+ * Contains Drupal\cloudflare\Form\CloudFlareAdminSettingsForm.
  */
 
 namespace Drupal\cloudflare\Form;
 
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
 use Drupal\Core\Form\ConfigFormBase;
 use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+use Drupal\cloudflare\CloudFlareStateInterface;
+use Drupal\cloudflare\CloudFlareZoneInterface;
+use Egulias\EmailValidator\EmailValidator;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use CloudFlarePhpSdk\ApiEndpoints\ZoneApi;
+use CloudFlarePhpSdk\Exceptions\CloudFlareException;
 
 /**
- * Class DefaultForm.
+ * Class CloudFlareAdminSettingsForm.
  *
  * @package Drupal\cloudflare\Form
  */
-class CloudFlareAdminSettingsForm extends ConfigFormBase {
+class CloudFlareAdminSettingsForm extends ConfigFormBase implements ContainerInjectionInterface {
```

Comment:
I really like this idea Niels. I am about to move config into Purge for Akamai. I want to get something usable out of Akamai first, but I am happy to try to attack a base form class once I am there. If you do go ahead and do something, let me know and I will move over to it.